platform/barefoot/sonic-platform-modules-bfn-montara/sonic_platform/platform_utils.py

```diff
@@ -18,5 +26,55 @@ def run_cmd(cmd):
         run_cmd(['mkdir', '-p', file_path])
     if not os.path.isfile(path):
         run_cmd(['touch', path])
-    if (mode is not None):    
+    if (mode is not None):
         run_cmd(['chmod', mode, path])
+
+def cancel_on_sigterm(func):
+    """
+    Wrapper for a function which has to be cancel on SIGTERM
+    """
+    @wraps(func)
```

Comment:
I have a concern about the 'signal handler backup/restore' this decorator:
1. Currently this code will work well because it's only 1 decorator. and seems there is no another code in this project to similar backup/restore.

2. However, because this is a decorator, so it's maybe use to another different method in future, then if 2 methods using this decorator running simultaneously, the signal handler will messed up by this decorator. here is a example:

step1: App using default signal handler handeler_D.

step1: Method1 start, signaler handler switch to handler_N, bacup handler is signal_D

step2: Method2 start, signal handler switch to handler_M, and backup handler will be handler_N

step3: Method1 end, signal handler restore to handler_D

step4: Method2 end, signal handler restore to handler_N, which is wrong because signal handler should be handler_D

I suggest define a global signal handler for this app, and also backup/restore the global signal handler.

Reply:
I agree that this scenario is possible in multi-threaded environment. But this is not a case with psud as of now. So, not sure whether it worth the efforts. Probably we can address this separately. What do you think?